src/clipboard/ClipboardManager.cpp

```diff
@@ -0,0 +1,92 @@
+#include "ClipboardManager.hpp"
+#include <hyprutils/os/Process.hpp>
+#include <hyprutils/string/String.hpp>
+#include <iterator>
+#include "../helpers/Log.hpp"
+#include <ranges>
+#include <sstream>
+
+using namespace Hyprutils::OS;
+
+struct SPipeDeleter {
+    void operator()(FILE* p) const {
+        if (p)
+            pclose(p);
+    }
+};
+
+CClipboardManager::CClipboardManager(const SClipboardConfig& config) : m_sConfig(config) {}
+
+static void executeCommand(const std::string& command, const std::string& arg) {
+    if (command.empty())
+        return;
+
+    std::string finalCommand   = command;
+    size_t      placeholderPos = finalCommand.find("{}");
```

Comment:
you are still doing this. Instead, you can make a dynamic call to std::vformat. Just keep in mind it can throw if the format is invalid